projects/dsp-ui/README.md

```diff
@@ -1,26 +1,288 @@
-# @knora/ui
+# @dasch-swiss/dsp-ui
 
-Reusable Angular elements for Angular applications based on knora-api.
+The modules help to create a graphical user interface, a web application to use [DSP-API](https://api.knora.org) in a quick and simple way. The modules are written in Typescript to use them with **[Angular](https://angular.io) (version 9)**. We decided to style components and directives with [Angular Material design](https://material.angular.io).
 
-This library was generated with [Angular CLI](https://github.com/angular/angular-cli) version 8.2.14.
+DSP-UI-LIB implements [DSP-JS-LIB](https://www.npmjs.com/package/@knora/api) to connect with the DSP API. DSP-API is a software framework for storing, sharing, and working with primary sources and data in the humanities.
 
-## Code scaffolding
+DSP-API and the DSP-UI-LIB are [free software](http://www.gnu.org/philosophy/free-sw.en.html), released under the [GNU Affero General Public](http://www.gnu.org/licenses/agpl-3.0.en.html).
 
-Run `ng generate component component-name --project @knora/ui` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module --project @knora/ui`.
-> Note: Don't forget to add `--project @knora/ui` or else it will be added to the default project in your `angular.json` file.
+This version of DSP-UI-LIB **requires [DSP-API version ^12.0.0](https://github.com/dasch-swiss/knora-api/releases/tag/v12.0.0).**
 
-## Build
+## Installation
 
-Run `ng build @knora/ui` to build the project. The build artifacts will be stored in the `dist/` directory.
+```bash
+npm install @dasch-swiss/dsp-ui
+```
 
-## Publishing
+### Dependencies
 
-After building your library with `ng build @knora/ui`, go to the dist folder `cd dist/knora/ui` and run `npm publish`.
+The module has the following package dependencies, which you also have to install.
 
-## Running unit tests
+<!-- TODO: the following package will be renamed to @dasch-swiss/dsp-js -->
```

Comment:
These are peer dependencies defined in the lib's `package.json`. 
No need to mention the exact versions here.

Reply:
they have to be met on the app level (app's `package.json`) -> `npm install`

Reply:
Yes I thought, it will install the dependencies. But this wasn't the case when I used the package in my own test app. But I will try again...

Reply:
you could list the main points of the roadmap if you want (future steps)

Reply:
peer deps are not installed with npm i

Reply:
> peer deps are not installed with npm i

is this a question or a statement?

Reply:
a statement

Reply:
Then I do not understand your comment. This README is the one from the lib and should help people who want to use the lib modules in their app. So they have to install the peer deps.
The difference between the two READMEs:
- main README in the root of the repo is for developers, who want help on developing the lib and the modules
- lib README will be published on npm and is for developers who want to use the DSP-UI-LIB in their app.

Reply:
> Then I do not understand your comment. This README is the one from the lib and should help people who want to use the lib modules in their app. So they have to install the peer deps.

What I mean is: just let the reader know that there are peer deps and refer to the lib's `package.json`. Why mention the exact versions in the README?

Reply:
I think this list is left from previous lib. The dependencies should be displayed on npm. But for some reasons, the PeerDeps are not listed there. We never updated this issue. I will have a look on it and remove the list from README.

Reply:
all the dependencies are listed in lib's package.json under `peerDependencies` which means: "The user of the library has to install the dependency (adding it to his own package.json)". I think we should list some specific or additional packages under `dependenices`. This will install the packages automatically when someone uses the library.

Reply:
> This will install the packages automatically when someone uses the library.

> Angular libraries should list all @angular/* dependencies as peer dependencies. This ensures that when modules ask for Angular, they all get the exact same module. If a library lists @angular/core in dependencies instead of peerDependencies, it might get a different Angular module instead, which would cause your application to break.
https://angular.io/guide/creating-libraries#use-typescript-path-mapping-for-peer-dependencies

I am not sure what the best practices are for non Angular deps.

Reply:
the `JDNConvertibleCalendarDateAdapter` needs to be a peer dependency because it has to be configured on the app level.

Reply:
@knora/api is also used at the app level, so it has to be a peer dep

Reply:
I think `"ngx-color-picker": "^9.1.0"` is missing form the lib's deps

Reply:
I guess you can only move a dep in the lib's dependency if it's use is encapsulated in the lib. This is the same situation we had with @knora/api and RxJS.

Reply:
I will make a DSP task for this.

Reply:
https://dasch.myjetbrains.com/youtrack/issue/DSP-232